### DISCUSSION TITLE
About the state of the implemented eCPA model

### DISCUSSION THREAD

**Discussion by tinibi98** [ClapeyronThermo/Clapeyron.jl#207] [Q&A]
I just started looking into modeling electrolytes with eCPA and was wondering whether the model that has been implemented on the Electrolytes branch worked for NaCl-H2O mixtures' osmotic coefficients and activity coefficients. Thanks a lot for this open source project!

**Comment by pw0908** [comment]
It's been a long while since we've modified the Electrolyte branch. Odds are, if you were to clone it on its own, it would work, but there's no guarantees...

I did implement and benchmark eCPA on that branch ages ago and that was for the NaCl+water system, so I know it works for that.

**Reply by tinibi98** [reply]
Alright, thank you. As I am new to ClapeyronThermo and to Julia, what would be the prompt to setup the eCPA Maribo-Mogensen (MM?) model and plot osmotic coefficients for the NaCl+water system?

**Reply by tinibi98** [reply]
I used an old commit (376fd0f , Oct 30 2021) to setup the eCPA model but it produces wrong osmotic coefficients for the NaCl-H2O system. Do you remember what properties you benchmarked against in the past and how I could possibly reproduce those results?

**Reply by pw0908** [reply]
Ill double check. I do recall that the eCPA folks were very imprecise when it came to describing their mixing rules and I got stuck trying to understand what they did. 

If you don't mind, Im going to go and bring the electrolyte branch up to speed with master.

**Reply by tinibi98** [reply]
I appreciate your help so so much! Is it the 2015 Maribo-Mogensen paper you based your implementation on?
[xiong2023.pdf](https://github.com/ClapeyronThermo/Clapeyron.jl/files/12747118/xiong2023.pdf)
I attached a very new paper that is based on MM 2015 but with a simplified description for the ion concentration dependant static permittivity. This is the one I tried to implement for the NaCl+H2O system but I wasn't able to compute correct osmotic coefficients. I would really like to keep using Clapeyron as I think it is pretty cool.

**Reply by pw0908** [reply]
That paper is quite a bit clearer than the original eCPA paper. And I now remember why I had such a hard time implementing the mixing rule: the unlike parameters are salt-based while all the other variables are ion-based. I remember working through the maths and finding that there are cases where the system of equations needed to solve for the ion-based unlike parameters becomes over specified. I have a feeling, if you check our database, the unlike parameters for Na and Cl are not the correct ones for the NaCl mixtures. 

Do you need to use eCPA? Or could you consider something like ePCSAFT or eSAFTVRMie?

**Reply by tinibi98** [reply]
That is what I thought when I found that paper. I do not need to use eCPA, it was just the first thing I discovered when it came to EoSs for electrolytes. Does ePCSAFT perform in a similar way? On the long run I would like to model gas solubilities with the eEoS. Can you recommend any paper to read into it so that I can use Clapeyron's ePCSAFT? Which commit should I try to use?

**Reply by pw0908** [reply]
ePCSAFT performs very similarly to eCPA. Im not quite sure which one has more parameters, but you can find the relevant literature here: https://www.sciencedirect.com/science/article/pii/S0263876214002469. I still need to look into our implementation to make sure we have the right one. I do know we have both SAFT-VRE Mie and eSAFT-VR Mie (you can find details on both here: https://www.sciencedirect.com/science/article/pii/S0378381221003198). 

In all of these approaches, you can model gas solubilities.

**Reply by tinibi98** [reply]
Thanks again, I will have a look into these recommendations. Once you figured out your implementation, it would be also very interesting to learn, how osmotic coefficients are computed with Clapeyron for these specfic eEoS. At this point I should probably introduce myself: My name is Tim and I am a PhD student from Germany who wants to model thermodynamically consistent fluid flows with the lattice Boltzmann method. This is why I am that interested.

**Reply by pw0908** [reply]
Nice to meet you Tim! Im hoping to get back to electrolytes after AIChE in Early November. I could've sworn I had implemented osmotic coefficients but it seems I only did so locally. Ill let you know once this has been implemented. 

Very brave of you to try and include these models within fluid flow simulations! Are you planning to implement this in Julia as well?

**Reply by tinibi98** [reply]
Looking forward to hearing from you in November! The LBM code I am using and also developing is an OpenSource project of my group called "OpenLB", a C++ library, see openlb.net .
